search/ranker.cpp

```diff
@@ -329,20 +329,74 @@ void Ranker::Init(Params const & params, Geocoder::Params const & geocoderParams
   m_tentativeResults.clear();
 }
 
-Result Ranker::MakeResult(RankerResult const & r) const
+Result Ranker::MakeResult(RankerResult const & rankerResult, bool needAddress,
+                          bool needHighlighting) const
 {
-  Result res = GenerateFinalResult(r, true /* needAddress */);
-  HighlightResult(m_params.m_tokens, m_params.m_prefix, res);
-  if (ftypes::IsLocalityChecker::Instance().GetType(r.GetTypes()) == ftypes::NONE)
+  ReverseGeocoder::Address addr;
+  bool addrComputed = false;
+
+  string name = rankerResult.GetName();
+  if (needAddress && name.empty())
+  {
+    // Insert exact address (street and house number) instead of empty result name.
+    if (!addrComputed)
+    {
+      m_reverseGeocoder.GetNearbyAddress(rankerResult.GetCenter(), addr);
+      addrComputed = true;
+    }
+    if (addr.GetDistance() == 0)
+      name = FormatStreetAndHouse(addr);
+  }
+
+  uint32_t const type = rankerResult.GetBestType(&m_params.m_preferredTypes);
+
+  // Format full address only for suitable results.
+  string address;
+  if (needAddress)
   {
-    m_localities.GetLocality(res.GetFeatureCenter(), [&](LocalityItem const & item) {
-      string city;
-      if (item.GetSpecifiedOrDefaultName(m_localityLang, city))
-        res.AppendCity(city);
-    });
+    address = rankerResult.GetRegionName(m_infoGetter, type);
+    if (ftypes::IsAddressObjectChecker::Instance()(rankerResult.GetTypes()))
+    {
+      if (!addrComputed)
+      {
+        m_reverseGeocoder.GetNearbyAddress(rankerResult.GetCenter(), addr);
+        addrComputed = true;
+      }
+      address = FormatFullAddress(addr, address);
+    }
+  }
+
+  // todo(@m) Used because Result does not have a default constructor. Factor out?
+  auto mk = [&](RankerResult const & r) -> Result {
+    switch (r.GetResultType())
+    {
+    case RankerResult::Type::TYPE_FEATURE:
+    case RankerResult::Type::TYPE_BUILDING:
+      return Result(r.GetID(), r.GetCenter(), name, address,
+                    m_categories.GetReadableFeatureType(type, m_params.m_currentLocaleCode), type,
+                    r.GetMetadata());
+    default:
```

Comment:
Please, replace default by:
```c++
case RankerResult::Type::TYPE_LATLON:
   ...
```
and move ASSERTION below the switch.

Reply:
done